File: minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.ts
function minStartValue(nums: number[]): number {
    let sum: number=0;
    let minVal: number = 0;
    
    for(const n of nums){
        sum+=n;
        minVal = Math.min(minVal, sum);
    }
    
    return 1-minVal;
};

File: minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.cs
public class Solution {
    public int MinStartValue(int[] nums) {        
        int minVal = 0;
        int sum = 0;
        
        foreach(int n in nums)
        {
            sum += n;
            minVal = Math.Min(minVal, sum);
        }
        
        return 1-minVal;
    }
}